.github/workflows/scripts/detect-all-changes.sh

```
@@ -0,0 +1,175 @@
+    continue
+  fi
+  
+  plugin_name=$(basename "$plugin_dir")
+  version_file="${plugin_dir}version"
+  
+  if [ ! -f "$version_file" ]; then
+    echo "   ⚠️ No version file for: $plugin_name"
+    continue
+  fi
+  
+  current_version=$(cat "$version_file" | tr -d '\n\r')
```

Shellcheck (0.10.0)

[style] Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

(SC2002)